### PR TITLE
Clndapi 779 fix parsing

### PR DESCRIPTION
**Pull request by antonpriyma**

### CODE REVIEW

File: calendar.go [hunk 1]
```diff
@@ -435,7 +435,7 @@ func ParseCalendar(r io.Reader) (*Calendar, error) {
 		}
 		line := ParseProperty(*l)
 		if line == nil {
-			return nil, errors.New("Error parsing line")
+			continue
```
Comment (antonpriyma) [inline calendar.go:438]: игнорим, если не получилось распарсить строку. В оутлуке часто кастомная фигня всякая

File: calendar.go [hunk 2]
```diff
@@ -534,7 +534,7 @@ func (cs *CalendarStream) ReadLine() (*ContentLine, error) {
 			}
 			if len(p) == 0 {
 				c = false
-			} else if p[0] == ' ' {
+			} else if p[0] == ' ' || p[0] == '\t' {
```
Comment (antonpriyma) [inline calendar.go:537]: По RFC https://icalendar.org/iCalendar-RFC-5545/3-1-content-lines.html

`Lines of text SHOULD NOT be longer than 75 octets, excluding the line break. Long content lines SHOULD be split into a multiple line representations using a line "folding" technique. That is, a long line can be split between any two characters by inserting a CRLF immediately followed by a single linear white-space character **(i.e., SPACE or HTAB). Any sequence of CRLF followed immediately by a single linear white-space character is ignored (i.e., removed) when processing the content type**.`

File: calendar.go (hunk 1 above)
Comment (arran4) [reply calendar.go:438]: Sorry. I can't read this.

File: calendar.go (hunk 2 above)
Comment (arran4) [reply calendar.go:537]: ✔️

File: calendar.go (hunk 1 above)
Comment (arran4) [reply calendar.go:438]: Can you provide a case this fixes or write a test to show that it doesn't break anything?

File: components.go [hunk 3]
```diff
@@ -603,7 +603,7 @@ func ParseComponent(cs *CalendarStream, startLine *BaseProperty) (ComponentBase,
 		}
 		line := ParseProperty(*l)
 		if line == nil {
-			return cb, errors.New("Error parsing line")
+			continue
```
Comment (arran4) [inline components.go:606]: This also. It feels like a large change?

File: calendar.go (hunk 2 above)
Comment (antonpriyma) [reply calendar.go:537]: Sorry for my Russian comments, i will change it to english)

File: components.go (hunk 3 above)
Comment (antonpriyma) [reply components.go:606]: Yes, you right. You lib does not support custom properties(https://datatracker.ietf.org/doc/html/rfc5545#section-3.8.8.2). I will try to support this

File: components.go [hunk 4]
```diff
@@ -603,6 +603,9 @@ func ParseComponent(cs *CalendarStream, startLine *BaseProperty) (ComponentBase,
 		}
 		line := ParseProperty(*l)
 		if line == nil {
+			if strings.HasPrefix(string(*l), "X-") {
+				continue
```
Comment (antonpriyma) [inline components.go:607]: as i mention before, your code fails is case of custom ics fields(https://icalendar.org/iCalendar-RFC-5545/3-8-8-2-non-standard-properties.html). However it`s not necessarily, so i think we can skip this properties in such way.

**Comment by arran4** [comment]
Can you rebase back into your branch?

**Comment by arran4** [comment]
Hey I saw you made another commit, is this ready for a review?

**Comment by arran4** [comment]
@antonpriyma Is this something you are still actively interested in continuing?

**Comment by arran4** [comment]
Hey @antonpriyma I believe I have managed to fix these issues elsewhere. The tests had an error in the ical examples which was causing the failure.

**Comment by arran4** [comment]
See https://github.com/arran4/golang-ical/pull/49

**Pull request closed without merging**